htlcswitch/switch.go

```diff
@@ -566,6 +577,49 @@ func (s *Switch) handlePacketForward(packet *htlcPacket) error {
 			return err
 		}
 
+		// First, we check to make sure we do not have an existing
+		// half-circuit add for this htlc. If one is found, this
+		// indicates that we already tried to forward this to a link,
+		// but the full circuit was never recorded. The incoming packet
+		// is likely to be the same packet begin replayed after startup,
+		// we conservatively cancel this back to make sure it gets
+		// cleared from commitment transactions along the path.
+		oldHalfCircuit := s.lookupHalfCircuit(packet.halfKey())
+		if oldHalfCircuit != nil {
+			// If packet was forwarded from another channel link
+			// than we should notify this link that some error
+			// occurred.
+			failure := lnwire.NewTemporaryChannelFailure(nil)
+			reason, err := packet.obfuscator.EncryptFirstHop(failure)
+			if err != nil {
+				err := errors.Errorf("unable to obfuscate "+
+					"error: %v", err)
+				log.Error(err)
+				return err
+			}
+
+			addErr := errors.Errorf("unable to forward packet after " +
+				"indeterminate forwarding decision")
+			log.Error(addErr)
+
+			if err = source.HandleSwitchPacket(&htlcPacket{
+				incomingChanID: packet.incomingChanID,
+				incomingHTLCID: packet.incomingHTLCID,
+				isRouted:       true,
```

Comment:
I think this should actually be _false_ as the packet doesn't have the outgoing htlc ID and channel ID set.